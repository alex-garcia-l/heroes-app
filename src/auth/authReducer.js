import { authType } from '../types/auth';

export const authReducer = (state = {}, action) => {
  
  switch (action.type) {
    case authType.login:
      return {
        ...action.payload,
        logged: true
      }
    case authType.logout:
      return {
        logged: false
      }
  
    default:
      return state;
  }

}