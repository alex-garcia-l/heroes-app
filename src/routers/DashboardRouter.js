import React from 'react';
import { Route, Routes } from 'react-router-dom';

import { DcScreen } from '../components/dc/DcScreen';
import { HeroScreen } from '../components/heros/HeroScreen';
import { HomeScreen } from '../components/HomeScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { SearchScreen } from '../components/search/SearchScreen';

import { Navbar } from '../components/ui/Navbar';

export const DashboardRouter = () => {
  return (
    <>
      <Navbar />
      <div className="container mt-4">
        <Routes>
          <Route exact index path="/" element={<HomeScreen />} />
          <Route exact path="marvel" element={<MarvelScreen />} />
          <Route exact path="search" element={<SearchScreen />} />
          <Route exact path="hero/:id" element={<HeroScreen />} />
          <Route exact path="dc" element={<DcScreen />} />
        </Routes>
      </div>
    </>
  );
};
