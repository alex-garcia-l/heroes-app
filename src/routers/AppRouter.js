import React from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';

import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRouter } from './DashboardRouter';

export const AppRouter = () => {
  return (
    <BrowserRouter>
      <Routes>

        <Route
          exact
          path="login"
          element={
            <PublicRoute>
              <LoginScreen />
            </PublicRoute>
          }
        />

        <Route path="/*" 
          element={
            <PrivateRoute>
              <DashboardRouter />
            </PrivateRoute>
          }
        />
      </Routes>
    </BrowserRouter>
  );
};
