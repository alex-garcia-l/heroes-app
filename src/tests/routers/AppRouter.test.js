import { mount } from 'enzyme';
import { AuthContext } from '../../auth/authContext';
import { AppRouter } from '../../routers/AppRouter';

describe('Prueba para el componente <AppRouter />', () => {
  
  test('debe mostrar el login si no está autenticado', () => {

    const contextValue = {
      user: {
        logged: false
      }
    }

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').text().trim()).toBe('Please sign in');
  });

  test('debe mostrar la página home si está autenticado', () => {

    const contextValue = {
      user: {
        name: 'Name',
        logged: true
      }
    }

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('header').exists()).toBe(true);
    expect(wrapper.find('span.fs-4').text().trim()).toBe('Heroes App');
    expect(wrapper.find('h2').first().text().trim()).toBe('Search');
  });
  

});
