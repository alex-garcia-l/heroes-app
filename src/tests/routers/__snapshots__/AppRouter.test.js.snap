// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba para el componente <AppRouter /> debe mostrar el login si no está autenticado 1`] = `
<div
  className="text-center d-flex align-items-center vh-100"
>
  <main
    className="form-signin"
  >
    <form
      onSubmit={[Function]}
    >
      <img
        alt=""
        className="mb-4"
        height="57"
        src="https://getbootstrap.com/docs/5.0/assets/brand/bootstrap-logo.svg"
        width="72"
      />
      <h1
        className="h3 mb-3 fw-normal"
      >
        Please sign in
      </h1>
      <button
        className="w-100 btn btn-lg btn-primary"
        type="submit"
      >
        Sign in
      </button>
      <p
        className="mt-5 mb-3 text-muted"
      >
        © 
        2022
      </p>
    </form>
  </main>
</div>
`;

exports[`Prueba para el componente <AppRouter /> debe mostrar la página home si está autenticado 1`] = `
Array [
  <header
    className="p-3 bg-dark text-white"
  >
    <div
      className="container"
    >
      <div
        className="d-flex flex-wrap align-items-center justify-content-center justify-content-lg-start"
      >
        <a
          className="d-flex align-items-center mb-2 mb-lg-0 text-white text-decoration-none me-0 me-md-4"
          href="/home"
          onClick={[Function]}
        >
          <span
            className="fs-4"
          >
            Heroes App
          </span>
        </a>
        <ul
          className="nav col-12 col-lg-auto me-lg-auto mb-2 justify-content-center mb-md-0"
        >
          <li>
            <a
              aria-current="page"
              className="nav-link px-2 text-secondary"
              href="/home"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li>
            <a
              className="nav-link px-2 text-white"
              href="/marvel"
              onClick={[Function]}
            >
              Marvel
            </a>
          </li>
          <li>
            <a
              className="nav-link px-2 text-white"
              href="/dc"
              onClick={[Function]}
            >
              Dc
            </a>
          </li>
        </ul>
        <div
          className="text-end"
        >
          <span
            className="me-4"
          >
            Name
          </span>
          <button
            className="btn btn-outline-light me-2"
            onClick={[Function]}
            type="button"
          >
            Logout
          </button>
        </div>
      </div>
    </div>
  </header>,
  <div
    className="container mt-4"
  >
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <h2>
          Search
        </h2>
        <hr />
      </div>
      <div
        className="col-12 col-sm-6 m-auto my-5"
      >
        <form
          className=""
          onSubmit={[Function]}
        >
          <div
            className="input-group"
          >
            <input
              autoComplete="off"
              className="form-control"
              name="inputSearch"
              onChange={[Function]}
              placeholder="Search a hero"
              type="text"
              value=""
            />
            <button
              className="btn btn-outline-primary"
              type="submit"
            >
              Search
            </button>
          </div>
        </form>
      </div>
      <div
        className="col-12"
      >
        <h2>
          Heros
        </h2>
        <hr />
        <div
          className="alert alert-info"
        >
          Search a hero
        </div>
        <div
          className="row"
        />
      </div>
    </div>
  </div>,
]
`;
